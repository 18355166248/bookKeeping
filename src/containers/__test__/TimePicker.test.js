import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import TimePciker from '../TimePicker/TimePicker'

let wrapper, props
describe('测试 Timepicker 组件', () => {
  beforeEach(() => {
    props = {
      year: '2019',
      month: '9',
      onChange: jest.fn()
    }
    wrapper = shallow(<TimePciker {...props} />)
  })
  it('匹配快照', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('年份渲染是否正确', () => {
    const Input = wrapper.find('Input')
    expect(Input.props().value).toEqual(`${props.year} 年 ${props.month} 月`)
  })
})
