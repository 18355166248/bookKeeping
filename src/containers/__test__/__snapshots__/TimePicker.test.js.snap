// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Timepicker 组件 匹配快照 1`] = `
<Dropdown
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={
    <Menu
      className="drop_down_menu"
    >
      <MenuItem
        className="time_picker_menu_item"
      >
        <div
          className="drop_down"
        >
          <div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2015
               年
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2016
               年
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2017
               年
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2018
               年
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2019
               年
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2020
               年
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2021
               年
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2022
               年
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2023
               年
            </div>
          </div>
          <div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              1
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              2
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              3
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              4
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              5
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              6
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              7
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              8
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              9
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              10
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              11
               月
            </div>
            <div
              className=" common_cell"
              onClick={[Function]}
            >
              12
               月
            </div>
          </div>
        </div>
      </MenuItem>
    </Menu>
  }
  placement="bottomLeft"
  trigger={
    Array [
      "click",
    ]
  }
>
  <Input
    className="btn "
    readOnly={true}
    suffix={
      <Icon
        type="calendar"
      />
    }
    type="text"
    value="2019 年 9 月"
  />
</Dropdown>
`;
